admin.php

```diff
@@ -140,6 +149,37 @@ function deleteMap () {
 	header("Location: admin.php?status=mapDeleted");
 }
 
+function flaggedComments() {
+	$results = array();
+	$data = MapComment::getAllFlagged();
+	$results['comments'] = $data['results'];
+	$results['pageTitle'] = "Flagged Comments | CTM Map Repository";
+
+	require($_SERVER['DOCUMENT_ROOT'] . TEMPLATE_PATH . "/admin/listComments.php");
+}
+
+function deleteComment() {
+	$comment = MapComment::getById($_GET['id']);
+	if (!$comment) {
+		header("Location: admin.php?error=commentNotFound");
+		return;
+	}
+
+	$comment->delete();
+	header("Location: admin.php?status=commentDeleted");
+}
+
+function restoreComment() {
+	$comment = MapComment::getById($_GET['id']);
+	if (!$comment) {
+		header("Location: admin.php?error=commentNotFound");
+		return;
+	}
+	$comment->flagCount = 0;
```

Comment:
I think I'd prefer another field on the comment for indicating "Admin Approved" instead of simply resetting the flag count, otherwise people could spam report a given person's comments endlessly and constantly remove them, even if an admin has already reviewed and approved the comment.

Reply:
That's a good idea. I've added a field on the comment in the database which gives administrators the official override on a comment so no matter how many times people will spam the report button, if you override, that's the final say.

See commit `b01dfb2` for the details.